pyiron_atomistics/vasp/vasprun.py

```diff
@@ -538,18 +537,108 @@ def parse_structure_to_dict(self, node, d):
 
+    def _parse_INCAR_generator_vector_parameters(self, val_type, val, filename, param_name):
+        """
+        Helper function to convert a Vasprun array-type parameter into the proper
+        type. Boolean, int and float types are converted.
+
+        Args:
+            val_type: Value type parsed from vasprun.xml.
+            val: Actual string value parsed for vasprun.xml.
+            filename: Fullpath of vasprun.xml. Used for robust error handling.
+                E.g., if vasprun.xml contains *** for some Incar parameters,
+                the code will try to read from an INCAR file present in the same
+                directory.
+            param_name: Name of parameter.
+
+        Returns:
+            Parsed value.
+        """
+        err = ValueError("Error in parsing vasprun.xml")
+        if val_type == "logical":
+            val = [i == "T" for i in val.split()]
+        elif val_type == "int":
+            try:
+                val = [int(i) for i in val.split()]
+            except ValueError:
+                # Fix for stupid error in vasprun sometimes which displays
+                # LDAUL/J as 2****
+                val = self._parse_from_incar(filename, param_name)
+                if val is None:
+                    raise err
+        elif val_type == "string":
+            val = val.split()
+        else:
+            try:
+                val = [float(i) for i in val.split()]
+            except ValueError:
+                # Fix for stupid error in vasprun sometimes which displays
+                # MAGMOM as 2****
+                # val = self._parse_from_incar(filename, param_name)
+                va
+                if val is None:
+                    raise err
+        return val
+
+    def _parse_INCAR_and_generator(self, elem):
+        params = {}
+        for c in elem:
+            name = c.attrib.get("name")
+            if c.tag not in ("i", "v"):
+                p = self._parse_INCAR_and_generator(c)
+                if name == "response functions":
+                    # Delete duplicate fields from "response functions",
+                    # which overrides the values in the root params.
+                    p = {k: v for k, v in p.items() if k not in params}
+                params.update(p)
+            else:
+                ptype = c.attrib.get("type")
+                val = c.text.strip() if c.text else ""
+                try:
+                    if c.tag == "i":
+                        params[name] = self._parse_INCAR_generator_parameters(ptype, val)
+                    else:
+                        params[name] = self._parse_INCAR_generator_vector_parameters(ptype, val, self.filename, name)
+                except Exception as exc:
+                    if name == "RANDOM_SEED":
+                        # Handles the case where RANDOM SEED > 99999, which results in *****
+                        params[name] = None
+                    else:
+                        raise exc
```

Comment:
```suggestion
                except Exception:
                    if name == "RANDOM_SEED":
                        # Handles the case where RANDOM SEED > 99999, which results in *****
                        params[name] = None
                    else:
                        raise
```
Using raise inside an except block re-raises automatically.  Can we also make the exception more specific?